pkgs/development/tools/misc/stm32cubemx/default.nix

```diff
@@ -1,47 +1,52 @@
-{ lib, stdenv, requireFile, makeDesktopItem, libicns, imagemagick, jre, fetchzip }:
-
+{ lib, stdenv, makeDesktopItem, icoutils, fdupes, imagemagick, jdk11, fetchzip }:
+# TODO: JDK16 causes STM32CubeMX to crash right now, so we fixed the version to JDK11
+# This may be fixed in a future version of STM32CubeMX. This issue has been reported to ST:
+# https://community.st.com/s/question/0D53W00000jnOzPSAU/stm32cubemx-crashes-on-launch-with-openjdk16
+# If you're updating this derivation, check the link above to see if it's been fixed upstream
+# and try replacing all occurrences of jdk11 with jre and test whether it works.
 let
-  version = "6.0.1";
+  version = "6.2.1";
   desktopItem = makeDesktopItem {
     name = "stm32CubeMX";
     exec = "stm32cubemx";
     desktopName = "STM32CubeMX";
     categories = "Development;";
     icon = "stm32cubemx";
   };
+  iconame = "STM32CubeMX";
 in
 stdenv.mkDerivation rec {
   pname = "stm32cubemx";
   inherit version;
 
 
   src = fetchzip {
```

Comment:
```suggestion
  version = "6.2.1";

  src = fetchzip {
```

Reply:
Can fetchzip still work in that case? It needs `version` to construct the download link.

Reply:
Also, I don't want to be pedantic, but do we really need to change much more about this package? Is it good practice to have as little `let` or `inherit` statements as possible?

Reply:
Ah right, it works because of `rec`. Done!